lint.go

```diff
@@ -750,6 +751,37 @@ func CheckEarlyDefer(f *lint.File) {
 	f.Walk(fn)
 }
 
+func CheckDubiousSyncPoolPointers(f *lint.File) {
+	fn := func(node ast.Node) bool {
+		call, ok := node.(*ast.CallExpr)
+		if !ok {
+			return true
+		}
+
+		sel, ok := call.Fun.(*ast.SelectorExpr)
+		if !ok {
+			return true
+		}
+		if sel.Sel.Name != "Put" {
+			return true
+		}
+		typ := f.Pkg.TypesInfo.TypeOf(sel.X)
+		if typ == nil || (typ.String() != "sync.Pool" && typ.String() != "*sync.Pool") {
```

Comment:
This is my proposed solution to permit both `sync.Pool` and `*.sync.Pool`, there's other ways to do this, but I thought this was relatively readable and straight forward, but it's also not perfectly tidy, so happy to hear alternatives.